Feature 3041 appendix a release guide

jprestop opened the pull request:
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Reviewed changes in RTD [here](https://metplus.readthedocs.io/en/feature_3041_appendix_a_release_notes/Release_Guide/#instructions-summary) for each components' development and official release instructions (Titled: Update Appendix A METplus Components Python Packages).

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

@georgemccabe, @bikegeek, and @JohnHalleyGotway Please review docs/Release_Guide/release_steps/update_appendix_a_python.rst carefully to ensure the instructions are clear with regard to what needs to be done.
I would like an approval from each of you (as opposed to only one of you) since this issue impacts everyone who regularly cuts a release. I would like to make sure the language is clear and makes sense from the perspective of all components.

Please note that while I updated appendixA.rst for this issue, it was updated earlier this summer and will still need to be updated for this release, as I noticed some changes in one of the requirements.txt file.  As such, this release is a good opportunity to ensure the instructions are clear and the updates are made.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Do these changes introduce new SonarQube findings? **[No]**</br>
If **yes**, please describe:

- [x] Please complete this pull request review by **[20250916]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table.
- [x] For any new datasets, an entry to the [METplus Verification Datasets Guide](https://metplus.readthedocs.io/en/latest/Verification_Datasets/index.html).
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **Coordinated METplus-X.Y Support** project for bugfix releases or **METplus-Wrappers-X.Y.Z Development** project for official releases
- [ ] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

jprestop committed:
Per #3041, updating appendix A

jprestop committed:
Merge branch 'develop' of github.com:dtcenter/METplus into feature_3041_appendix_a_release_notes

jprestop committed:
Per #3041, updating release notes to reflect need to update Appendix A

georgemccabe committed:
update link to MET Python Embedding reqs to use latest instead of feature branch

georgemccabe reviewed (dismissed):
These changes look good. I made an update to fix the link to the MET Python Embedding requirements which contained a feature branch. Thanks for making these changes!

JohnHalleyGotway commented on docs/Release_Guide/release_steps/update_appendix_a_python.rst:
```diff
@@ -0,0 +1,23 @@
+  and their version number (if applicable) should be updated.
+
+  The versions of the Python packages should match the versions listed in the
+  requirements.txt file at top level of each repository and should include any new
```
MET has no `requirements.txt` file at the top-level directory of the repository. The `docs/requirements.txt` file lists the reqs for Read-The-Docs. Are the Python requirements for MET actually defined anywhere in the code itself? Or are the Python embedding requirements listed exclusively in the documentation? If that latter, we should have a custom version of these instructions for MET to clarify that point.

georgemccabe replied:
@JohnHalleyGotway, we could add a `requirements.txt` file in the MET repository that lists the requirements for Python Embedding.

jprestop replied:
@JohnHalleyGotway I would like to add a requirements.txt in the MET repository that lists the requirements for Python Embedding based on @georgemccabe's suggestion. That adds consistency. However, I can see a counterpoint that Python Embedding is _optional_ functionality, so perhaps a requirements.txt file is misleading. I still lean toward adding it. What are your thoughts?

JohnHalleyGotway replied:
I believe that `requirements.txt` files define dependencies for Python packages. MET, itself, is obviously not a Python package, but we do use `scripts/python/met` like a package.

So perhaps we put it in [scripts/python/met](https://github.com/dtcenter/MET/tree/main_v12.1/scripts/python/met) or somewhere close to that? I'd recommend that @DanielAdriaansen and/or @hsoh-u weigh in on where this belongs.

JohnHalleyGotway commented on docs/Release_Guide/release_steps/update_appendix_a_python.rst:
```diff
@@ -0,0 +1,23 @@
+  and their version number (if applicable) should be updated.
+
+  The versions of the Python packages should match the versions listed in the
+  requirements.txt file at top level of each repository and should include any new
```
```suggestion
  :code:`requirements.txt` file at top level of each repository and should include any new
```
I like formatting the file names as "code" to make them stand out. But if there's some other convention we use, use that instead.

jprestop replied:
Thanks @JohnHalleyGotway.  I also like that formatting, and I appreciate you catching that the filename was not formatted in that way.

jprestop commented:
> These changes look good. I made an update to fix the link to the MET Python Embedding requirements which contained a feature branch. Thanks for making these changes!

@georgemccabe Thank you for the review and for making that update to fix the link to the MET Python Embedding requirements (that was strange!). I appreciate you taking care of that problem.

jprestop committed:
Update docs/Release_Guide/release_steps/update_appendix_a_python.rst

Co-authored-by: John Halley Gotway <johnhg@ucar.edu>

JohnHalleyGotway commented on docs/Release_Guide/release_steps/update_appendix_a_python.rst:
```diff
@@ -0,0 +1,23 @@
+  and their version number (if applicable) should be updated.
+
+  The versions of the Python packages should match the versions listed in the
+  :code:`requirements.txt` file at top level of each repository and should include any new
```
```suggestion
  :code:`requirements.txt` file(s) within each repository and should include any new
```

jprestop replied:
Perfect! Thanks!

jprestop committed:
Update docs/Release_Guide/release_steps/update_appendix_a_python.rst

Co-authored-by: John Halley Gotway <johnhg@ucar.edu>

bikegeek commented:
@jprestop

1. For METcalcpy, under section 3.3, there is a typo (environ**n**ent instead of environ**m**ent):

`Please refer to https://docs.conda.io/projects/conda/en/latest for more information abount conda as a package and environnent manager.`

2. Can you remove the

`Note: In a future release, METcalcpy will be located on PyPI (Python Package Index) to facilitate installation into a virtual environment or conda environment using `pip <packagename>. ``

It is on PyPI, but I haven't been updating it.

bikegeek commented:
@jprestop: 

METdataio is OK

bikegeek commented:
@jprestop for METplotpy

**Section 3.1.3 Install METplotpy in a conda environment**
I misspelled 'dependencies' twice (as dependancies) in :
```
This will install METplotpy into the conda env, along with all the dependancies listed above in requirements.txt.

If you already have an environment setup, or want to install METplotpy without the dependancies, add the –no-deps argument to pip.
```

**Section 3.1.5**
the bash command is missing a ':' between the last path and ${PYTHONPATH}:

`$METCALCPY_SOURCE/METcalcpy:$METCALCPY_SOURCE/METcalcpy/util:$METPLOTPY_SOURCE/METplotpy${PYTHONPATH}`

it should be $METCALCPY_SOURCE/METcalcpy:$METCALCPY_SOURCE/METcalcpy/util:$METPLOTPY_SOURCE/METplotpy:${PYTHONPATH}

**Section 3.1.6** 
is a bit out of date with respect to only the performance diagram as the only plot written using Matplotlib:

`These plots were written using Python plotly, with the exception of the performance diagram, which was written using Matplotlib.` 

It is probably easier to remove this sentence, as it doesn't provide any useful information to the user with respect to setting up or running.

bikegeek commented:
@jprestop for METviewer

Section 3.2.1 Configure and Build METviewer

Under item 1. Configure the batch and loading tools:

the last bullet can be removed, this step is not necessary:

```
Edit METviewer/webapp/metviewer/WEB-INF/classes/log4j.properties:

        Set log4j.appender.logfile.File setting to the absolute path of a log file
```

jprestop commented:
> @jprestop
> 
> 1. For METcalcpy, under section 3.3, there is a typo (environ**n**ent instead of environ**m**ent):
> 
> `Please refer to https://docs.conda.io/projects/conda/en/latest for more information abount conda as a package and environnent manager.`
> 
> 2. Can you remove the
> 
> `Note: In a future release, METcalcpy will be located on PyPI (Python Package Index) to facilitate installation into a virtual environment or conda environment using `pip . ``
> 
> It is on PyPI, but I haven't been updating it.

Hi @bikegeek I don't see these in the [files I changed](https://github.com/dtcenter/METplus/pull/3101/files) for this PR. I only touched appendixA.rst in the User's Guide and added release_steps/update_appendix_a_python.rst in the Release Guide (plus the includes in the Release Guide for this new release step). Are these references in one of those .rst files?

jprestop commented:
> @jprestop for METplotpy
> 
> **Section 3.1.3 Install METplotpy in a conda environment** I misspelled 'dependencies' twice (as dependancies) in :
> 
> ```
> This will install METplotpy into the conda env, along with all the dependancies listed above in requirements.txt.
> 
> If you already have an environment setup, or want to install METplotpy without the dependancies, add the –no-deps argument to pip.
> ```
> 
> **Section 3.1.5** the bash command is missing a ':' between the last path and ${PYTHONPATH}:
> 
> `$METCALCPY_SOURCE/METcalcpy:$METCALCPY_SOURCE/METcalcpy/util:$METPLOTPY_SOURCE/METplotpy${PYTHONPATH}`
> 
> it should be $METCALCPY_SOURCE/METcalcpy:$METCALCPY_SOURCE/METcalcpy/util:$METPLOTPY_SOURCE/METplotpy:${PYTHONPATH}
> 
> **Section 3.1.6** is a bit out of date with respect to only the performance diagram as the only plot written using Matplotlib:
> 
> `These plots were written using Python plotly, with the exception of the performance diagram, which was written using Matplotlib.`
> 
> It is probably easier to remove this sentence, as it doesn't provide any useful information to the user with respect to setting up or running.


Hi @bikegeek I don't see these in the [files I changed](https://github.com/dtcenter/METplus/pull/3101/files) for this PR. I only touched appendixA.rst in the User's Guide and added release_steps/update_appendix_a_python.rst in the Release Guide (plus the includes in the Release Guide for this new release step). Are these references in one of those .rst files?

jprestop commented:
> @jprestop for METviewer
> 
> Section 3.2.1 Configure and Build METviewer
> 
> Under item 1. Configure the batch and loading tools:
> 
> the last bullet can be removed, this step is not necessary:
> 
> ```
> Edit METviewer/webapp/metviewer/WEB-INF/classes/log4j.properties:
> 
>         Set log4j.appender.logfile.File setting to the absolute path of a log file
> ```


Hi @bikegeek I don't see these in the [files I changed](https://github.com/dtcenter/METplus/pull/3101/files) for this PR. I only touched appendixA.rst in the User's Guide and added release_steps/update_appendix_a_python.rst in the Release Guide (plus the includes in the Release Guide for this new release step). Are these references in one of those .rst files?

bikegeek commented:
@jprestop, sorry about that.  Disregard the comments above, I'll focus on the files you changed.

bikegeek reviewed (approved):
METcalcpy, METdataio, METplotpy, and METviewer changes are good.

jprestop commented:
> @jprestop, sorry about that. Disregard the comments above, I'll focus on the files you changed.

That's ok @bikegeek.  It looks like we have some other changes that would be good to make in the documentation... Unfortunately, I don't have any funding to fix those things now.
